src/sc/sidechaintypes.h

```diff
@@ -4,198 +4,252 @@
 
 ////////////////////////////// Custom Config types //////////////////////////////
-class CustomFieldConfig
+class CustomCertificateFieldConfig
 {
 public:
-    CustomFieldConfig() = default;
-    virtual ~CustomFieldConfig() = default;
-    virtual int32_t getBitSize() const = 0;
+    CustomCertificateFieldConfig() = default;
+    virtual ~CustomCertificateFieldConfig() = default;
+    virtual bool IsValid() const = 0;
 };
 
-class CompressedFieldElementConfig : public CustomFieldConfig
+class FieldElementCertificateFieldConfig : public CustomCertificateFieldConfig
 {
 private:
     int32_t nBits;
-    bool isBitsLenghtValid(); //TENTATIVE IMPLEMENTATION, BEFORE ACTUAL ONE
 
 public:
-    CompressedFieldElementConfig(int32_t nBitsIn);
-    CompressedFieldElementConfig(): CustomFieldConfig(), nBits(0) {} //for serialization only, which requires the default ctor
-    ~CompressedFieldElementConfig() = default;
+    FieldElementCertificateFieldConfig(int32_t nBitsIn);
+    FieldElementCertificateFieldConfig(const FieldElementCertificateFieldConfig& rhs) = default;
+    ~FieldElementCertificateFieldConfig() = default;
+
+    //For serialization only, which requires the default ctor. No checkValid call here
+    FieldElementCertificateFieldConfig(): CustomCertificateFieldConfig(), nBits(0) {}
 
-    int32_t getBitSize() const override; //TENTATIVE IMPLEMENTATION, BEFORE ACTUAL ONE
+    bool IsValid() const override final;
 
     ADD_SERIALIZE_METHODS;
     template <typename Stream, typename Operation>
     inline void SerializationOp(Stream& s, Operation ser_action, int nType, int nVersion) {
         READWRITE(nBits);
-
-        if (!isBitsLenghtValid())
-            throw std::invalid_argument("CompressedFieldElementConfig size must be strictly positive");
     }
 
-    bool operator==(const CompressedFieldElementConfig& rhs) const {
+    int32_t getBitSize() const;
+
+    bool operator==(const FieldElementCertificateFieldConfig& rhs) const {
         return (this->nBits == rhs.nBits);
     }
 
-    bool operator!=(const CompressedFieldElementConfig& rhs) const {
+    bool operator!=(const FieldElementCertificateFieldConfig& rhs) const {
         return !(*this == rhs);
     }
 
-    friend std::ostream& operator<<(std::ostream& os, const CompressedFieldElementConfig& r) {
+    friend std::ostream& operator<<(std::ostream& os, const FieldElementCertificateFieldConfig& r) {
         os << r.nBits;
         return os;
     }
 };
 
-class CompressedMerkleTreeConfig : public CustomFieldConfig
+class BitVectorCertificateFieldConfig : public CustomCertificateFieldConfig
 {
 private:
-    int32_t treeHeight;
-    static const int32_t MAX_TREE_HEIGHT = log2(std::numeric_limits<int32_t>::max()); //TENTATIVE IMPLEMENTATION, BEFORE ACTUAL ONE
-    bool isTreeHeightValid(); //TENTATIVE IMPLEMENTATION, BEFORE ACTUAL ONE
+    int32_t bitVectorSizeBits;
+    int32_t maxCompressedSizeBytes;
```

Comment:
I can't get why we need two values? Is it not enough to have only `maxCompressedSizeBytes`?

Reply:
Discussed AFK. Done